galley/pkg/server/callout.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/authplugin"
+	"istio.io/istio/galley/pkg/authplugins"
+	"istio.io/istio/pkg/mcp/source"
+)
+
+type callout struct {
+	address string
+	so      *source.Options
+	do      []grpc.DialOption
+	cancel  context.CancelFunc
+}
+
+// Test override types
+type dialFn func(string, ...grpc.DialOption) (*grpc.ClientConn, error)
+type mcpClient interface {
+	Run(context.Context)
+}
+type newClientFn func(mcp.ResourceSinkClient, *source.Options) mcpClient
+type closeFn func(*grpc.ClientConn)
+
+// Test override variables
+var (
+	connClose       closeFn
+	grpcDial        dialFn
+	sourceNewClient newClientFn
+)
+
+func init() {
+	// Setup non-test defaults
+	grpcDial = grpc.Dial
```

Comment:
Consider following the patch table model of server code.

Reply:
Done